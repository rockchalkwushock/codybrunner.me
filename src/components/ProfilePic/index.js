import styled from 'styled-components'

import profilePic from '../../assets/profile_pic.jpg'

/**
 * @fileOverview
 * Profile Pic
 *
 * REVIEW
 * I would like to look at the perf diff of
 * - importing into CSS (webpack)
 * - `gatsby-image` utilizing the bells & whistles of `sharp`.
 */

const ProfilePic = styled.div`
  background-color: ${({ theme }) => theme.site.bg};
  background-image: url(${profilePic});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 50%;
  height: 200px;
  margin-left: auto;
  margin-right: auto;
  width: 200px;
`

ProfilePic.displayName = 'ProfilePic'

export default ProfilePic
