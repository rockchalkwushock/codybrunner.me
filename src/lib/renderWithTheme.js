import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
