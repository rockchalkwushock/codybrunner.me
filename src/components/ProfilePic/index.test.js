import React from 'react'

import renderWithTheme from '../../lib/renderWithTheme'
import ProfilePic from './'

describe('Component: <ProfilePic />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<ProfilePic />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
