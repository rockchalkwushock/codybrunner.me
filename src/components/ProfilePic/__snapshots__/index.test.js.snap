// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <ProfilePic /> should render correctly 1`] = `
.c0 {
  background-color: #7a9eb1;
  background-image: url();
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 50%;
  height: 200px;
  margin-left: auto;
  margin-right: auto;
  width: 200px;
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "fg": "",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
      },
    }
  }
>
  <ProfilePic>
    <div
      className="c0"
    />
  </ProfilePic>
</ThemeProvider>
`;
